packages/sdk/src/utils/CachingMap.ts

```diff
@@ -18,14 +24,11 @@ export class CachingMap<K extends MapKey, V, P extends any[]> {
 
     private readonly cachedFn: (...args: P) => Promise<V>
     private readonly cache: LRU<K, { data: V, maxAge: number }>
+    private readonly opts: Options<P, K>
```

Comment:
(nitpick) A `Readonly<>` here could provide more clarity to the reader that the options are not meant to change during the life of an instance of `CachingMap`.
```suggestion
    private readonly opts: Readonly<Options<P, K>>
```

Reply:
We haven't used this kind of deep annotation in the `sdk` code base before.

Many of our objects are immutable, but adding many `Readonly<>` annotations to all of those would just decrease readability. Using `readonly` in the field type is not the same, but it communicates the intention quite well.

(Also the name and the type of the field should communicate clearly that we are using the Configuration Option Pattern which implies immutability by default.)